programs/survival/bucketstack.sc

```diff
@@ -0,0 +1,133 @@
+__config() ->
+(
+	//change these number to whatever you want the maximum stack size to be
+	//note that the max scarpet can inventory_set() is 2^31 - 1
+	global_buckets = m(
+		l('water',1),
+		l('lava',1),
+		l('milk',1),
+		l('pufferfish',1),
+		l('salmon',1),
+		l('cod',1),
+		l('tropical_fish',1),
+		l('axolotl',1)
+	);
+	return(
+		m(
+			l('scope','player'),
+			l('stay_loaded',true)
+		)
+	);
+);
+
+__on_player_collides_with_entity(player,entity) ->
+(
+	if(entity~'pickup_delay' == 0,
+		l(item,count,nbt) = entity~'item';
+		if(item~'_bucket',
+			//if it's an item we're picking up that is a non-empty bucket
+			//first determine the max stack size
+			type = null;
+			while(type == null,length(global_buckets),
+				type = item~(keys(global_buckets):_);
+			);
+			max_stack = global_buckets:type;
+			//then find a non-full slot to put it in
+			//modified lines 103 and 104 of gnembon's shulkerboxes.sc to find a non-full slot
+			slot = -1;
+			while(count && (slot = inventory_find(player,item,slot+1)) != null && slot < 36,inventory_size(player)-5,

```

Comment:
Itn's this avoiding the offhand too?

Reply:
As far as I'm aware, no item ever goes into the offhand on pickup, not even a shield. Thus, the offhand should be avoided.

Reply:
Not even to stack?

Reply:
You were right; if there was already a partial stack, the offhand will take the items that are picked up. Now the script matches the vanilla behavior.